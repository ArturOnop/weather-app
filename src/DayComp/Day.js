import React from 'react';
import db from "../db.json"
import getDate from "./dateTransform.js";

function Day(weather) {

    let minTemp = typeof weather["minTemp"] === "number" ? weather["minTemp"].toString() : "N/A";
    let maxTemp = typeof weather["maxTemp"] === "number" ? weather["maxTemp"].toString() : "N/A";
    let pressure = typeof weather["pressure"] === "number" ? weather["pressure"].toString() : "N/A";
    let humidity = typeof weather["humidity"] === "number" ? weather["humidity"].toString() : "N/A";
    let uvIndex = typeof weather["uvIndex"] === "number" ? weather["uvIndex"].toString() : "N/A";
    let windSpeed = typeof weather["windSpeed"] === "number" ? weather["windSpeed"].toString() : "N/A";
    let windDir = typeof weather["windDir"] === "string" ? weather["windDir"] : "N/A";
    let imgSrc = db.img[weather["overall"]] ? db.img[weather["overall"]] : "";

    let date;
    try {
        date = getDate(weather.date);
    } catch (e) {
        date = "N/A"
        console.error(e);
    }

    return (
        <div className="day">
            <div className="date">{date}</div>
            <img className="overall" src={imgSrc} alt="weather-icon"/>
            <div className="temp">
                From <div className="minTemp">{minTemp}</div> C°
                To <div className="maxTemp">{maxTemp}</div> C°
            </div>
            <div className="pressure">Pressure: {pressure} mBar</div>
            <div className="humidity">Humidity: {humidity}%</div>
            <div className="uvIndex">UV Index: {uvIndex}</div>
            <div className="windSpeed">Wind speed: {windSpeed} km\hour</div>
            <div className="windDir">Wind direction: {windDir}</div>
        </div>
    );
}

export default Day;
