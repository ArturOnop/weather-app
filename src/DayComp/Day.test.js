import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Day from './Day.js';

Enzyme.configure({adapter: new Adapter()});

describe("Valid Day", () => {
    let wrapper;
    let dayExample = {
        "date": "14/02/2022",
        "minTemp": -4,
        "maxTemp": -1,
        "pressure": 1011,
        "humidity": 81,
        "uvIndex": 0,
        "windSpeed": 21,
        "windDir": "north",
        "overall": "snow"
    };

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<Day {...dayExample}/>);
    });

    test("should contain classname day", () => {
        expect(wrapper.find(".day").exists()).toBe(true);
        expect(wrapper.find(".day").length).toBe(1);
    });

    test("should contain classname date with provided date", () => {
        expect(wrapper.find(".date").exists()).toBe(true);
        expect(wrapper.find(".date").length).toBe(1);
        expect(wrapper.find(".date").text()).toBe("Monday 14 february 2022 year");
    });

    test("should contain classname overall with provided imgSrc", () => {
        expect(wrapper.find(".overall").exists()).toBe(true);
        expect(wrapper.find(".overall").length).toBe(1);
        expect(wrapper.find(".overall").prop("src"))
            .toBe("https://img.icons8.com/small/32/000000/snow.png");
    });

    test("should contain classname minTemp and maxTemp with provided data", () => {
        expect(wrapper.find(".minTemp").exists()).toBe(true);
        expect(wrapper.find(".maxTemp").exists()).toBe(true);

        expect(wrapper.find(".minTemp").length).toBe(1);
        expect(wrapper.find(".maxTemp").length).toBe(1);

        expect(wrapper.find(".minTemp").text()).toBe("-4");
        expect(wrapper.find(".maxTemp").text()).toBe("-1");
    });

    test("should contain classname pressure with provided pressure", () => {
        expect(wrapper.find(".pressure").exists()).toBe(true);
        expect(wrapper.find(".pressure").length).toBe(1);
        expect(wrapper.find(".pressure").text()).toBe("Pressure: 1011 mBar");
    });

    test("should contain classname humidity with provided humidity", () => {
        expect(wrapper.find(".humidity").exists()).toBe(true);
        expect(wrapper.find(".humidity").length).toBe(1);
        expect(wrapper.find(".humidity").text()).toBe("Humidity: 81%");
    });

    test("should contain classname uvIndex with provided uvIndex", () => {
        expect(wrapper.find(".uvIndex").exists()).toBe(true);
        expect(wrapper.find(".uvIndex").length).toBe(1);
        expect(wrapper.find(".uvIndex").text()).toBe("UV Index: 0");
    });

    test("should contain classname windSpeed with provided windSpeed", () => {
        expect(wrapper.find(".windSpeed").exists()).toBe(true);
        expect(wrapper.find(".windSpeed").length).toBe(1);
        expect(wrapper.find(".windSpeed").text()).toBe("Wind speed: 21 km\\hour");
    });

    test("should contain classname windDir with provided windDir", () => {
        expect(wrapper.find(".windDir").exists()).toBe(true);
        expect(wrapper.find(".windDir").length).toBe(1);
        expect(wrapper.find(".windDir").text()).toBe("Wind direction: north");
    });

    test("should match snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});

describe("Invalid Day", () => {
    let wrapper;
    let dayExample = {
        "date": 12062022,
        "minTemp": "invalid",
        "maxTemp": "invalid",
        "pressure": "invalid",
        "humidity": "invalid",
        "uvIndex": "invalid",
        "windSpeed": "invalid",
        "windDir": 123,
        "overall": "invalid"
    };

    global.console = {error: jest.fn()};

    beforeEach(() => {
        wrapper = shallow(<Day {...dayExample}/>);
    });

    test("should contain classname date with N/A", () => {
        expect(console.error).toBeCalled();
        expect(wrapper.find(".date").text()).toBe("N/A");
    });

    test("should contain classname overall with no src", () => {
        expect(wrapper.find(".overall").prop("src")).toBe("");
    });

    test("should contain classname minTemp and maxTemp with N/A", () => {
        expect(wrapper.find(".minTemp").text()).toBe("N/A");
        expect(wrapper.find(".maxTemp").text()).toBe("N/A");
    });

    test("should contain classname pressure with N/A", () => {
        expect(wrapper.find(".pressure").text()).toBe("Pressure: N/A mBar");
    });

    test("should contain classname humidity with N/A", () => {
        expect(wrapper.find(".humidity").text()).toBe("Humidity: N/A%");
    });

    test("should contain classname uvIndex with N/A", () => {
        expect(wrapper.find(".uvIndex").text()).toBe("UV Index: N/A");
    });

    test("should contain classname windSpeed with N/A", () => {
        expect(wrapper.find(".windSpeed").text()).toBe("Wind speed: N/A km\\hour");
    });

    test("should contain classname windDir with N/A", () => {
        expect(wrapper.find(".windDir").text()).toBe("Wind direction: N/A");
    });
});
