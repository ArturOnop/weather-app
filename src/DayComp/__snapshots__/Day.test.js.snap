// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Valid Day should match snapshot 1`] = `"<div class=\\"day\\"><div class=\\"date\\">Monday 14 february 2022 year</div><img class=\\"overall\\" src=\\"https://img.icons8.com/small/32/000000/snow.png\\" alt=\\"weather-icon\\"/><div class=\\"temp\\">From <div class=\\"minTemp\\">-4</div> C° To <div class=\\"maxTemp\\">-1</div> C°</div><div class=\\"pressure\\">Pressure: 1011 mBar</div><div class=\\"humidity\\">Humidity: 81%</div><div class=\\"uvIndex\\">UV Index: 0</div><div class=\\"windSpeed\\">Wind speed: 21 km\\\\hour</div><div class=\\"windDir\\">Wind direction: north</div></div>"`;
